packages/cms-data-sync/src/calendars/sync-all.ts

```diff
@@ -0,0 +1,46 @@
+import { Logger } from '@asap-hub/server-common';
+import { CalendarDataProvider } from '@asap-hub/model';
+
+import { fetchContentfulEntries, getContentfulClient } from '../utils';
+import { rateLimiter } from '../utils/rate-limiter';
+
+export async function syncCalendars({
+  dataProvider,
+  logger,
+}: {
+  dataProvider: CalendarDataProvider;
+  logger: Logger;
+}) {
+  const { contentfulEnvironment } = await getContentfulClient();
+
+  const calendars = await fetchContentfulEntries(
+    contentfulEnvironment,
+    'calendars',
+  );
+
+  if (!calendars.length) {
+    logger.info('No calendars to sync');
+    return;
+  }
+
+  const promises = await Promise.allSettled(
+    calendars.map(async (calendar) => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      await dataProvider.update(calendar.sys.id, { googleApiMetadata: {} });
+      await rateLimiter.removeTokens(3);
```

Comment:
This approach to rate limiting is probably redundant given the client is already [applying the same rate limits in the API adapter](https://github.com/yldio/asap-hub/blob/master/packages/cms-data-sync/src/utils/setup.ts#L21-L26).

However, if you are going to do it this way then you definitely need to remove the tokens _before_ making the update call, otherwise this map will fire off all the updates simultaneously without any rate limiting, and only rate limit the promise resolution _afterwards_.